lib/zendesk_apps_tools/server.rb

```diff
@@ -1,47 +1,65 @@
 require 'sinatra/base'
-require 'zendesk_apps_support/package'
+require 'zendesk_apps_support/installed'
 
 module ZendeskAppsTools
   class Server < Sinatra::Base
     set :protection, :except => :frame_options
-    set :public_folder, proc { "#{settings.root}/assets" }
     last_mtime = Time.new(0)
     ZENDESK_DOMAINS_REGEX = /^http(?:s)?:\/\/[a-z0-9-]+\.(?:zendesk|zopim|zd-(?:dev|master|staging))\.com$/
+    last_domain = nil
 
     get '/app.js' do
       access_control_allow_origin
       content_type 'text/javascript'
 
-      if File.exists? settings.config
-        curr_mtime = File.stat(settings.config).mtime
-        if curr_mtime > last_mtime
-          settings_helper = ZendeskAppsTools::Settings.new
-          settings.parameters = settings_helper.get_settings_from_file(settings.config, settings.manifest)
-          last_mtime = curr_mtime
+      settings_helper = ZendeskAppsTools::Settings.new
+
+      appsjs = []
+      installations = []
+      order = {}
+
+      settings.apps.each_with_index do |app, index|
+        package = app[:package]
+        app_id = installation_id = -(index+1)
+        app_name = package.manifest_json['name'] || 'Local App'
+
+        appsjs << package.compile_js(
+          app_name: app_name,
+          app_id: app_id,
+          assets_dir: "http://localhost:#{settings.port}/#{app_id}/",
+          locale: params['locale']
+        )
+
+        if app[:settings_file_path]
```

Comment:
Can you extract this to a separate method and perhaps add a comment explaining what it's doing